### PR TITLE
BaSD-MS-49: Create applications view

### PR DESCRIPTION
**Pull request by ravegliamartin**
### Entity: Applications

Functionalities Added:

- List Applications
- Create Applications
- Delete Applications

### CODE REVIEW

File: public/js/applicationForm.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+const interviewInput = document.getElementById('interview');
+const resultInput = document.getElementById('result');
+
+const form = document.getElementById('form');
+const saveButton = document.getElementById('saveButton');
+const errorMessage = document.getElementById('error_message');
+
+saveButton.disabled = !!applicationId;
+
+const dateFormat = (date) => `${[`0${date.getMonth() + 1}`.slice(-2),
```
Comment (NicoCagnina) [inline public/js/applicationForm.js:18]: Wouldn't it be simpler if you just used the date from the interview?

Comment (NicoCagnina) [inline public/js/applicationForm.js]: Wouldn't it be better if you just used the date from the interview?

File: public/js/applicationForm.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      });
+    })
+    .catch((error) => {
+      errorMessage.innerText = error;
+    });
+};
+
+fillSelect('/positions', positionInput);
+fillSelect('/postulants', postulantInput);
+fillSelect('/interviews', interviewInput);
```
Comment (estefrare) [inline public/js/applicationForm.js]: You are creating a reusable function but part of the logic is tied to the possible implementations.

```suggestion
const fillSelect = (url, parent, makeText) => {
  fetch(`${window.location.origin}/api${url}`)
    .then((response) => {
      if (response.status !== 200) {
        return response.json().then(({ message }) => {
          throw new Error(message);
        });
      }
      return response.json();
    })
    .then((response) => {
      saveButton.disabled = false;
      response.data.forEach((el) => {
        const option = document.createElement('option');
        // eslint-disable-next-line no-underscore-dangle
        option.value = el._id;
        option.innerText = makeText(el);
        parent.append(option);
      });
    })
    .catch((error) => {
      errorMessage.innerText = error;
    });
};

fillSelect('/positions', positionInput, (el) => `${el.firstName} ${el.lastName}`);
fillSelect('/postulants', postulantInput, (el) => el.jobDescription);
fillSelect('/interviews', interviewInput, (el) => el.date.slice(0, 19).replace('T', ' '));
```

File: public/views/applicationForm.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    </header>
+    <div class="container">
+      <h2 class="header">Applications Form</h2>
+      <form id="form">
+        <label for="position">Position:</label>
+        <select id="position" name="position" required></select>
+        <label for="postulant">Postulant:</label>
+        <select id="postulant" name="postulant" required></select>
+        <label for="interview">Interview:</label>
+        <select id="interview" name="interview"></select>
```
Comment (estefrare) [inline public/views/applicationForm.html:37]: see how a date is handled [here](https://github.com/BaSD-2021/Esteban-MindSet-server/pull/45/files#diff-efdbd9ee8c4229255aa369e839207b2f380fec7b405af942426dd30a7bb255f1R23)

File: public/js/applicationList.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        button.onclick = (e) => deleteApplication(application, e);
+        // eslint-disable-next-line no-underscore-dangle
+        tr.onclick = () => openEditApplication(application._id);
+        tr.append(jobDescriptionTD, postulantTD, interviewTD, resultTD, button);
+        tableContent.append(tr);
+      });
+    });
+};
+
+proceedBtn.onclick = () => {
```
Comment (estefrare) [inline public/js/applicationList.js:66]: update variable name based on the button content

File: public/js/applicationList.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+const proceedBtn = document.getElementById('procced-button');
+const navButton = document.getElementById('applicationsNav');
+const addButton = document.getElementById('addApplication');
+const tableContent = document.getElementById('table-content');
+
+let applicationId;
+
+const openEditApplication = (id) => {
+  // eslint-disable-next-line no-underscore-dangle
+  window.location.href = `${window.location.origin}/views/applicationForm.html?_id=${id}`;
```
Comment (estefrare) [inline public/js/applicationList.js:11]: it is not necessary to use a `_` in the query param name. It is better If you can avoid a lint disable command

**Review by estefrare — CHANGES_REQUESTED** [review]

**Review by estefrare — APPROVED** [review]

**Pull request merged**